Add testing template

MichaelClifford opened the pull request:
This PR adds a testing template we can use to build our e2e tests for the distributed workloads/CodeFlare Stack.

* The template includes an empty function for each of the test issues we outlined on our [issues](https://github.com/opendatahub-io/distributed-workloads/issues) page. But feel free to supplement with more supporting functions if needed. 

* I added a small subset of tests for the KubeRay operator to get us started and provide an example of how to write tests in this format. 

* There is also a `resources` directory where we can add whatever scripts, yaml files, etc we will need during testing. It is currently just populated with template python file.

Maxusmusti commented on tests/basictests/distributed-workloads.sh:
```diff
@@ -0,0 +1,57 @@
+uninstall_codeflare_operator
+
+
+os::test::junit::declare_suite_end
```
missing newline at EOF

Maxusmusti commented on tests/resources/test_mcad_ray.py:
```diff
@@ -0,0 +1,7 @@
+
+
+def test_sdk_functionality():
+    pass
```
missing newline at EOF

Maxusmusti commented on tests/basictests/distributed-workloads.sh:
```diff
@@ -0,0 +1,57 @@
+    haeder: "Installing Codeflare Operator"
+}
+
+function install_distributred_workloads_kfdef(){
```
distributred -> distributed

Maxusmusti reviewed (changes requested):
Some very minor nits, once fixed it LGTM

anishasthana commented on tests/basictests/distributed-workloads.sh:
```diff
@@ -0,0 +1,57 @@
+os::test::junit::declare_suite_start "$MY_SCRIPT"
+
+function install_codeflare_operator() {
+    haeder: "Installing Codeflare Operator"
```
```suggestion
    header: "Installing CodeFlare Operator"
```

MichaelClifford commented:
:upside_down_face: Sorry for the spelling errors. Should be fixed now.

Maxusmusti reviewed (approved):
LGTM

tedhtchang commented:
which function will test the Notebook ?

MichaelClifford commented:
We might need an additional tests for the notebook image itself.  However, I'm assuming it will be part of both `tests_mcad_ray_functionality()` and `tests_mcad_torchx_functionality()` as those will run the SDK from the notebook image.

KPostOffice commented on tests/basictests/distributed-workloads.sh:
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+source $TEST_DIR/common
```
Do you know where this actually comes from? Is this included in the image or something. I saw it in the odh-manifests too. I assume it provides the functions used for testing like `os::cmd::expect_success`

MichaelClifford replied:
I think you are right, @KPostOffice. you can see more about the testing framework here: https://github.com/opendatahub-io/odh-manifests/blob/master/tests/TESTING.md

KPostOffice reviewed (commented):
This looks good to me. Not blocking, just a question.

KPostOffice commented on tests/basictests/distributed-workloads.sh:
```diff
@@ -0,0 +1,57 @@
+
+
+install_codeflare_operator
+install_distributred_workloads_kfdef
```
```suggestion
install_distributed_workloads_kfdef
```

MichaelClifford commented:
@anishasthana @KPostOffice @tedhtchang @Maxusmusti 
I added a test README along with util file needed for testing and simplified the example tests. 

Please take another look and let me know if its good to go :smile:

jbusche commented:
LGTM, thanks Michael!
